content/src/content/jcr_root/apps/core/wcm/components/download/v1/download/clientlibs/editor/js/download.js

```diff
@@ -35,37 +35,51 @@
         var dialogContent = $dialogContent.length > 0 ? $dialogContent[0] : undefined;
 
         if (dialogContent) {
-            titleTuple = new CheckboxTextfieldTuple(dialogContent, titleCheckboxSelector, titleTextfieldSelector);
-            descriptionTuple = new CheckboxTextfieldTuple(dialogContent, descriptionCheckboxSelector, descriptionTextfieldSelector, true);
-            $cqFileUpload     = $dialog.find(".cq-FileUpload");
-            $cqFileUploadEdit = $dialog.find(".cq-FileUpload-edit");
-            if ($cqFileUpload) {
-                $cqFileUpload.on("assetselected", function(e) {
-                    fileReference = e.path;
-                    retrieveDAMInfo(fileReference).then(
-                        function() {
-                            titleTuple.reinitCheckbox();
-                            descriptionTuple.reinitCheckbox();
-                        }
-                    );
-                });
-                $cqFileUpload.on("click", "[coral-fileupload-clear]", function() {
-                    titleTuple.reset();
-                    descriptionTuple.reset();
+
+            var rteInstance = $(descriptionTextfieldSelector).data("rteinstance");
+            // wait for the description textfield rich text editor to signal start before initializing.
+            if (rteInstance && rteInstance.isActive) {
+                init($dialog, dialogContent);
+            } else {
+                $(descriptionTextfieldSelector).on("editing-start", function() {
+                    init($dialog, dialogContent);
```

Comment:
Won't this lead to multiple initialisations? If so, we should keep track if the initialisation already happened to avoid doing it multiple times.

Reply:
I tested it on localhost and cloud and can confirm that it only initializes once.